src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java

```diff
@@ -302,28 +304,168 @@ public void setOption(String optionStr) {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final DetailAST parent = ast.getParent();
-        // we do not want to check colon for cases and defaults
-        if (parent.getType() != TokenTypes.LITERAL_DEFAULT
-                && parent.getType() != TokenTypes.LITERAL_CASE) {
-            final String text = ast.getText();
-            final int colNo = ast.getColumnNo();
-            final int lineNo = ast.getLineNo();
-            final String currentLine = getLine(lineNo - 1);
+        if (isTargetNode(ast)) {
+            if (option == WrapOption.NL && isNewLineViolation(ast)) {
```

Comment:
please make it like:
`if (option == WrapOption.NL && isNewLineModeViolation(ast)) {`
the same method name update for other mode.

Reply:
done